airlock/business_logic.py

```diff
@@ -20,6 +17,10 @@
 from airlock.users import User
 
 
+# We use PurePosixPath as a convenient URL path representation (we reassign rather than
+# use `import as` to satisfy mypy that we intend to export this name)
+UrlPath = PurePosixPath
```

Comment:
Could we use [`NewType`](https://docs.python.org/3/library/typing.html#typing.NewType) here?

Reply:
Oh yes, good suggestion.

Reply:
It turns out we can't, for reasons I don't fully understand. However I've added an alternative implementation which does effectively the same thing.